### PR TITLE
#include guards now start with CARAT

### LINKED ISSUE #37: Change #include guards to not use _ prefix
Most (all?) #include guards in carat currently are prefixed by an underscore; but symbols starting with a single underscore are reserved. We should instead use a prefix like `CARAT_`. E.g. this:
```
#ifndef _LINUX_LIST_H
#define _LINUX_LIST_H
```
should become
```
#ifndef CARAT_LIST_H
#define CARAT_LIST_H
```

### PR DESCRIPTION
**Pull request by DominikBernhardt**
Closes #37 

I noticed that some #include guards end with a `_` and some don't. Is there a reason for this? Otherwise I'll unify that as well and update this PR.

### CODE REVIEW

**Comment by fingolfin** [comment]
There is no reason, please remove the trailing underscores as well, thanks

**Comment by DominikBernhardt** [comment]
Done.

**Comment by DominikBernhardt** [comment]
@fingolfin Sorry, my mistake. I experimented with clang-format and apparently forgot to switch branches. This is fixed now.

File: include/graph.h [hunk 1]
```diff
@@ -2,8 +2,8 @@
    extern "C" {
 #endif
 
-#ifndef _CARAT_GRAPH_H_
-#define _CARAT_GRAPH_H_
+#ifndef _CARAT_GRAPH_H
+#define _CARAT_GRAPH_H
```
Comment (fingolfin) [inline include/graph.h]: Remove leading underscore?

File: include/list.h [hunk 2]
```diff
@@ -3,8 +3,8 @@ extern "C" {
 #endif
 
 
-#ifndef _LINUX_LIST_H
-#define _LINUX_LIST_H
+#ifndef CARAT_LINUX_LIST_H
+#define CARAT_LINUX_LIST_H
```
Comment (fingolfin) [inline include/list.h]: Remove LINUX?

File: include/tsubgroups.h [hunk 3]
```diff
@@ -3,8 +3,8 @@ extern "C" {
 #endif
 
 
-#ifndef _TSUBGROUPS_H_
-#define _TSUBGROUPS_H_
+#ifndef CARAT__TSUBGROUPS_H
+#define CARAT__TSUBGROUPS_H
```
Comment (fingolfin) [inline include/tsubgroups.h]: Double underscore?

**Comment by DominikBernhardt** [comment]
Good point. Squashed into two commits.

**Pull request merged**